### PR TITLE
refactor: use new async await APIs and update ipfsd-ctl

### PR DESCRIPTION
**Pull request by alanshaw**
resolves https://github.com/ipfs/interop/issues/92 - does not specify `bits` anymore
resolves https://github.com/ipfs/interop/issues/31 - uses "test" profile for all daemon spawns
resolves https://github.com/ipfs/interop/issues/30 - new ipfsd-ctl exposes daemon logs via `DEBUG=ipfsd-ctl:daemon:std*`

TODO:

1. [x] fix the websocket-star tests - websocket-star was not converted to async/await and has a security vulnerability. Switch to using stardust or webrtc-star for the tests
1. [x] fix ipfsd-ctl to auto kill process if it does not stop within a timeout. The previous version of ipfsd-ctl had this feature and was killing go-ipfs daemons. I've observed here and in master that go-ipfs does not quit when running tests where it sends files to/from a js-ipfs node. @stebalien were you aware?
  * https://github.com/ipfs/js-ipfsd-ctl/pull/441
1. [ ] fix issue with ky never returning from a call to `res.json()` if the payload is BIG. IDK why but ky clones the response, which causes it to be piped to two places, the first is never consumed so for large responses the high water mark is hit and the stream stops flowing. It's similar to the problem we [worked around here](https://github.com/ipfs/js-ipfs-http-client/blob/d7eb0e8ffb15e207a8a6062e292a3b5babf35a9e/src/lib/error-handler.js#L12-L23).
  * https://github.com/ipfs/interop/pull/96/files#diff-6581b67864c5dbbaa734e512057741ceR163-R167

### CODE REVIEW

**Comment by jacobheun** [comment]
> fix the websocket-star tests - websocket-star was not converted to async/await and has a security vulnerability. Switch to using stardust or webrtc-star for the tests

I'll work on converting this over to webrtc-star

**Comment by Stebalien** [comment]
>  I've observed here and in master that go-ipfs does not quit when running tests where it sends files to/from a js-ipfs node. @Stebalien were you aware?

Go-ipfs will _try_ to wait for all in-progress commands to stop. If that doesn't work, it'll die after 10s.

We try to cancel in-progress commands but could be getting stuck. Could you file an issue somewhere? We should try killing go-ipfs with SIGQUIT sometime to see where it's getting stuck (that will dump a stack trace).

**Comment by hugomrdias** [comment]
@Stebalien should we wait for the 10s or just SIGKILL please have a look at https://github.com/ipfs/js-ipfsd-ctl/pull/441#pullrequestreview-347184019

**Comment by achingbrain** [comment]
Looks like it just needs the two gh urls removing from package.json and it's good to go?

@alanshaw build is green - is the fix for the `ky` issue mentioned at the top still required?

File: package.json [hunk 1]
```diff
@@ -40,30 +35,32 @@
...
+    "libp2p-webrtc-star": "^0.17.3",
     "mocha": "^5.2.0",
     "multihashes": "~0.4.14",
     "p-retry": "^4.1.0",
     "pretty-bytes": "^5.1.0",
     "promisify-es6": "^1.0.3",
     "random-fs": "^1.0.3",
-    "readable-stream-buffer-stream": "~0.0.2",
-    "rimraf": "^2.6.3"
+    "readable-stream-buffer-stream": "github:alanshaw/readable-stream-buffer-stream#feat/browser-friendly",
```
Comment (achingbrain) [inline package.json]:
```suggestion
    "readable-stream-buffer-stream": "^1.0.0",
```

**Comment by alanshaw** [comment]
> @alanshaw build is green - is the fix for the `ky` issue mentioned at the top still required?

The test is skipped. It still needs fixing (i.e. remove ky from ipfs-http-client) but we can probably merge this, open an issue and revisit when we've fixed the ky issue.

cc @hugomrdias

File: package.json (hunk 1 above)
Comment (alanshaw) [reply package.json]: Pushed this update (needs package-lock.json update as well) - thanks.

**Comment by hugomrdias** [comment]
> > @alanshaw build is green - is the fix for the `ky` issue mentioned at the top still required?
> 
> The test is skipped. It still needs fixing (i.e. remove ky from ipfs-http-client) but we can probably merge this, open an issue and revisit when we've fixed the ky issue.
> 
> cc @hugomrdias

yes, ill remove ky asap and come back to this issue.

**Comment by achingbrain** [comment]
I'm going to merge this since this branch is basically defacto master right now.

I've opened https://github.com/ipfs/interop/issues/101 to track the skipped test.

**Pull request merged**